### ISSUE TITLE
WarmRestartHook wrong relational operator?

### ISSUE THREAD

**Issue by BioFreak95** [atomistic-machine-learning/schnetpack#147]
Hey,
in the function 'WarmRestartHook', self.best_current is initialized with infinity.
on_validation_end it calls an if-condition, when the val_loss is bigger than best_current. This can not be reached. I think it should be val_loss is smaller than best_current.
Than there is a condition, where it asks if best_current is bigger than best_previous. I think also here it should be reverse.

Maybe I misunderstood some things. In this case, I would be very happy for a short explanation.

I have 2 more questions, which I want to put in this ticket. I hope you can give your opinions.
1. does it make sense to combine warmResarts with ReduceLROnPlateauHook?
2. Did you have experience with a huge amount of space, which is necessary? Have small areas of proteins (10x10x10a) with cutoff 6a. 
Schnet has n_atom_basis 64, filters 64, n_interactions 10. A batchsize bigger than 2 seems to big for a 8GB 1080GTX. An increment of the network to 128/128 with Batch_size 1 seems to be the maximum. Have you experience with this and an idea how to handle this? Batch_size of 1 or 2 seems very small. Did I assume a to big network or cutoff maybe? Maybe you can share your experience, which could help me at tuning.

**Comment by ktschuett** [comment]
You're right the first one should be reversed, otherwise it does not stop. The second one is correct, since the waiting time should be increased if the loss did not improve and reset otherwise (but I think it is better to change it to >=). I'll do a pull request.

To the questions:
1. You could try that. I found that training with WarmRestarts can lead to overfitting, since your increasing the weights drastically at the large learing rates after the restarts. Maybe the PlateauReduction may help.
2. Yes, that seems reaaly large. Some hints: 
- Try smaller cutoffs between 4-5A. That is often sufficient, at least for energy + forces.
- Try 3-6 interaction layers.
- If you did not, use the new TorchEnvironmentProvider:
```
data = AtomsData(..., environment_provider=TorchEnvironmentProvider(cutoff, device)
```
The default SimpleEnvironentProvider always calculates the full distance matrix and should only be used for small molecules. Alternatively, you can use the AseEnvironmentProvider, but the new one should be better.
- You can use the DataParallel layer from pytorch to train on multiple GPUs to multiply the batch size.

I hope that helps.

**Comment by BioFreak95** [comment]
Ah. Yeah. Now I see. The second one is not wrong. Sorry. 
But you are right, I think also that >= would be better.

OK. Thank you for the hints. I'll try them. :)
